### DISCUSSION TITLE
Can't resolve whilst offline error

### DISCUSSION THREAD

**Discussion by Tom-G-2010** [cockpit-project/cockpit#21436] [Q&A]
I setup a server running Ubuntu 24.04.1 LTS and installed Cockpit version 314-1. 
I logged in got the "Loading Available Updates Failed" message  and "Cannot refresh cache whilst offline"
I searched the documentation and several forums including this one.  I found two related items and tried both but neither resolved the issue and both made matters worse.

First I found the fix on the cockpit project site FAQ that detailed a creating a placeholder file and fake network interface.
I did this and the result was that the network manager app would no longer start.
I found information about reinstalling and restarting it and did so but that did not resolve the issue

Second I found the install instructions on the cockpit site that said to use the backport to ensure I was installing the latest version.  I did so.
I logged in and again got the "Loading Available Updates Failed" message  and "Cannot refresh cache whilst offline"
I was hoping that with the new install the placeholder file and fake network interface fix I would be back in business.  But, it did not work I was back to the network manager app no longer starting.

If I've missed a post somewhere with the answer I apologize, it wasn't for lack of searching and reading.  A link to such a post would be greatly appreciated.

I'm a novice at best with all this and I don't know where to go from here.  Any help or advice on how to troubleshoot this would be greatly appreciated. 

thanks!
Tom G.

**Comment by fafnir1** [comment]
Hi Tom, 
I ran into the same thing and applied the fix you mentioned from the FAQ.  It looks like there are 3 steps. I tried the first step which didn't resolve the issue, skipped the second since it appears to be related to raspberry pi and then tried the 3rd step (which I will copy below) and that seems to have resolved the issue  on my system. Ubuntu Server 24.10

3. Set up a “dummy” network interface:
nmcli con add type dummy con-name fake ifname fake0 ip4 1.2.3.4/24 gw4 1.2.3.1

**Comment by Krishna-Noutiyal** [comment]
It’s 2025, and I’m still dealing with the same issue on my Ubuntu installation. There doesn’t seem to be any official fix yet. Creating a dummy interface works as a workaround, but it’s not a real solution. For now, I’ll just have to settle for it, I suppose.

**Comment by Ainz** [comment]
Its 2026 and this is still an issue.

<img width="300" height="" alt="image" src="https://github.com/user-attachments/assets/cf2fcd4b-0156-4e7a-b234-9ffa6c819eff" />

 
I do not really get if it is Cockpit or Ubuntu problem but annoying as h*ll and has been like this for a decade or more.  

Incredible one has to resort to these type of fixes. But it works.

**Comment by Venefilyn** [comment]
See [this comment about the reason](https://github.com/cockpit-project/cockpit/issues/16963#issuecomment-1136711525), but there is a workaround on our website

https://cockpit-project.org/faq.html#error-message-about-being-offline

**Reply by arxeiss** [reply]
I have followed the FAQ. Did step 1 and step 3 (I don't have RaspberryPi). It worked well until I shutdown my server and started again.

My internet stopped working. I was able to connect to Cockpit, from my computer (local home network). But my server has no access to internet. I tried AI to help me, but result was the fake connection stopped working. So I cannot access Software updates anymore.

Any hint how to make work the fake connection and also internet?

**Comment by Ainz** [comment]
ok Freya, that is true, but I never got passed the Ubuntu installation guide, https://cockpit-project.org/running.html#ubuntu 
Forgot to check the FAQ, but I assumed there would be a link to it in the relevant documentation section... ah, it is at the bottom, not in the menu. ok... props for invisibility perhaps...

**Reply by Venefilyn** [reply]
Forgot to link here for others to see (know you saw it already), but created an issue to address this being confusing

- https://github.com/cockpit-project/cockpit-project.github.io/issues/852

**Comment by Chefty** [comment]
Based on what was said in the FAQ link above and some more commands I could fix the Software updates error page.
Here are the command lines I used instead:
1. `echo -e "[keyfile]\nunmanaged-devices=none" | sudo tee /etc/NetworkManager/conf.d/10-globally-managed-devices.conf`
2. `sudo nmcli con add type dummy con-name fake ifname fake0 ip4 192.0.2.2/24` (removed the last part with gw4)
3. `nano /etc/netplan/50-cloud-init.yaml`

```
network:
  renderer: NetworkManager <----- Add this line
  ethernets:
    enp3s0:
      dhcp4: true
  version: 2
```

5. `sudo netplan apply`
6. reboot

Now I have a green checkmark saying ✅ **_System is up to date_**

p.s. you might want to run at the end `sudo apt upgrade -y` to force it and not wait.

**Reply by Venefilyn** [reply]
> 4\. Add one line right under network `renderer: NetworkManager`

WDYM with this? An empty line?

What does it accomplish?

**Reply by Chefty** [reply]
Empty line? No I added this exact line to tell Netplan to hand over control to NetworkManager since it appears to me Cockpit only speaks to NetworkManager.

```
network:
  renderer: NetworkManager <----- I added this line here
  ethernets:
    enp3s0:
      dhcp4: true
  version: 2
```

**Reply by Venefilyn** [reply]
Ahh sorry I thought it was "add one line underneath the 'renderer: NetworkManager'". Thanks!
